docs/reduction-in-locks.md

```diff
@@ -0,0 +1,106 @@
+{ .power-number }
+
+1. Set a backup lock on the server to prevent any DDL operations, such as creating or altering tables.
+
+2. Parse and copy the redo logs from the checkpoint up to the current `Log Sequence Number` (LSN), and start tracking new file operations.
+
+3. Identify and copy the `.ibd` files.
+
+4. Copy `non-InnoDB` files.
+
+5. Gather a sync point from all engines (InnoDB LSN, binlog, Global Transaction Identifier (GTID), etc.) by querying the log status.
+
+6. Stop the redo thread once it copies at least up to sync point.
+
+7. Release the backup lock on the server.
+
+If the backup lock is disabled with the [`--lock-ddl=OFF`](./xtrabackup-option-reference.md/#lock-ddl) option, a backup continues while concurrent DDL operations are executed. These backups may be invalid and may fail at either the `backup` or the `--prepare` step.
+
+## How the `--lock-ddl=REDUCED` option works
+
+Taking backup with the `--lock-ddl=REDUCED` option includes the following phases:
+
+### Phase 1: Operations performed without the lock
+
+!!! warning
+
+    The `ALTER INSTANCE ROTATE MASTER KEY` command should not be executed during Phase 1 if there are encrypted tables, as it will cause the backup to fail.
+
+The following operations are performed without the lock:
+{ .power-number }
+
+1. Parse and copy the redo logs from the checkpoint up to the current `LSN`, and start tracking new file operations.
+2. Start the redo log thread to copy the redo logs. 
+3. Track file operations by parsing the `MLOG_FILE_*` records from the redo log. These records help track changes in the files being backed up to ensure consistency.
+4. Copy the `.ibd` files.
+
+### Phase 2: Operations performed under the lock
+
+The following operations are performed under the lock:
+{ .power-number }
+
+1. Set the backup lock on the server to prevent new DDL operations, such as creating or altering tables.
+2. Check the file operations that were tracked and recopy the tablespaces.
+3. Create additional `metadata` files to perform the required actions (deletions or renames) on the already copied files. This approach ensures that the backup remains consistent and accurate without disrupting the streaming process. This is particularly important for streaming backups.
+4. Gather a sync point from all engines (InnoDB LSN, binlog, GTID, etc.) by querying the `log_status`.
+5. Stop the redo thread once it copies at least up to sync point at step 4.
+6. Release the backup lock on the server.
+
+??? note "The list of metadata files"
+
+    | File         | Description                                                                                                      |
+    |--------------|------------------------------------------------------------------------------------------------------------------|
+    | `.new`       | For files that need to be recopied due to encryption or `ADD INDEX` operations.                                  |
```

Comment:
For new files that are created while the backup in progress and also for files that need to be recopied due to..